### PR TITLE
Update developer docs for 0.4 (Part 3)

### LINKED ISSUE #1208: Improve developer docs
The new docs format is a joy. The recent hackathon shed light on a few aspects of our setup that cause problems for first-time contributors. 

### Documentation expansion
- [x] explanation of the automated tests (both travis and snap)
- [x] useful commands for rebuilding packages
- [ ] ~document need to build ossec packages for staging~ (no longer necessary; see #1668)
- [x] useful commands for running tests locally
- [x] clearer descriptions of the staging environments
- [ ] ~explain use of `vagrant-digitalocean` plugin for remote testing~ (no longer necessary; see #1666)
- [x] document management scripts for `pip` requirements (see https://github.com/freedomofpress/securedrop/pull/1348#issuecomment-240582768)
- [x] explicit version recommendations for:
  - [x] vagrant
  - [x] ansible
  - [x] virtualbox
- [x] document CI pipeline (especially any manual actions required by FPF staff)

### Documentation reformatting
- [x] breakout the environment descriptions into separate page
- [x] clean up .travis.yml file (linting only)
- [x] consistently make `code` blocks copypastable (no prefixed `$`)
- [x] update "copyright" year in docs
- [x] add SD logo to navbar

Right now much of the developer docs are folded into a **Getting Started** section. Breaking out the environment descriptions to a dedicated page would make the docs easier to navigate.

### PR DESCRIPTION
**Pull request by conorsch**
## Status

Ready for review.

## Description of Changes

Fixes #1208. Completes the remainder of outstanding developer-oriented documentation related to 0.4 changes, particular the new CI pipeline (#1666).

Changes proposed in this pull request:

* Removes dangling logic referring to old SnapCI setup. These changes should have been cleaned up in #1666, but were missed during review. To wit:
  * Vagrantfile references to digitalocean plugin
  * Vagrantfile references to snapci host (for local testing)
  * Ansible role for configuring snapci host
  * .gitignore entries related to snapci secrets
* Greatly expands dev docs with regard to testing. Three documents now explain the setup:
  * Testing: Application Tests
  * Testing: Configuration Tests
  * Testing: CI

The CI docs are mostly included for posterity, and currently are relevant only to FPF staff. Still, it's better to have the docs than not, and I tried to strike a balance between transparency and frankly stating "You probably don't need this."

## Testing

Build the docs locally via `make docs` and browse through all the developer documentation end-to-end. Pay particular attention to the testing breakdown. If anything doesn't make sense, holler, and we can revise prior to merge. 

In particular I'd like another set of eyes on the **Application Tests** document; it feels a bit light to me, but more detail might be unnecessary.

Most of the lines _removed_ in the diff should be easy to review, since they are obviously outdated and unused. The lines _added_ in the diff are new content and deserve more careful scrutiny.

## Deployment

No concerns for deployment; these changes apply only to the developer environment and related documentation.

## Checklist

Testing N/A; we need human eyeballs on these changes to confirm they work as intended. Do check that the CI checks pass, particularly the docs linting step.

### CODE REVIEW

**Comment by redshiftzero** [comment]
Built these and they are looking good. Just gonna run through the CI docs with EC2 creds prior to :+1:ing

File: docs/development/testing_application_tests.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+.. _Pytest: https://docs.pytest.org/en/latest/
+.. _Selenium: http://docs.seleniumhq.org/docs/
+.. _Coveralls: https://github.com/coveralls-clients/coveralls-python
+
+Installation
+------------
+
+The application tests are installed automatically in the Development
+and App-Staging VMs, based on the contents of
```
Comment (redshiftzero) [inline docs/development/testing_application_tests.rst]: nit: 
"Development VM" -> "development VM" 
"App-Staging VMs" -> "app-staging VMs"

(just for consistency)

**Comment by redshiftzero** [comment]
Also is it expected that people will be using Ansible 2+ to use the EC2 workflow? Might want to add a note to the docs if so

File: docs/development/testing_application_tests.rst (hunk 1 above)
Comment (conorsch) [inline docs/development/testing_application_tests.rst]: Great point, I was frustrated by my own lack of convention here. Should probably add a note to the style guide about that.

**Comment by conorsch** [comment]
Yes, folks definitely need to use Ansible v2—technically Ansible v2 is now required for even local staging (although it's broken pending a fix for #951). We haven't documented the Ansible v2 requirement yet since we still need to shake out the details in #1146.

**Pull request merged**